src/machine/state/mod.rs

```diff
@@ -16,6 +17,49 @@ pub struct State {
 
 impl State {
     pub fn new() -> Self {
-        unimplemented!();
+        State {
+            // for the moment start execution at address 0
+            pc: 0,
+
+            sr: SRegisters::new(),
+            gpr: GPRegisters::new(),
+
+            // for the moment create 1MB of memory
+            mem: Memory::with_capacity(0x100000)
+        }
+    }
+
+    /// Display the state partially on the command line for testing purposes
+    pub fn debug_output(&self) {
```

Comment:
Yes, I REALLY do think we need to pass parameters, like e.g. how many registers the user would like to have displayed, what part of memory they would like have displayed. And I don't think that there is a way to do this when implementing fmt::Display, fmt::Debug etc. If you see a way, let me know :-]